social_panel/lib/twitter_client.rb

```diff
@@ -0,0 +1,37 @@
+# frozen_string_literal: true
+
+require 'faraday'
+require 'JSON'
+require 'pry'
+
+class TwitterClient
+
+  # URL = 'https://api.twitter.com/1.1'
+  # TOKEN = "Bearer #{ENV.fetch('SOCIAL_PANEL_TWITTER_ACCESS_TOKEN')}"
+
+  def connection
+      Faraday.new(
+        url: 'https://api.twitter.com/1.1',
+        headers: { 'Authorization' => "Bearer #{ENV['SOCIAL_PANEL_TWITTER_ACCESS_TOKEN']}" },
+      )
+  end
+
+  def get_tweet_by_id(tweet_id)
+    response = connection.get("statuses/show.json?id=#{tweet_id}")
```

Comment:
Looks much better!

You can still do one improvement. The documentations shows this example:

```ruby
url = 'http://sushi.com/nigiri/sake.json'
resp = Faraday.get(url, {a: 1}, {'Accept' => 'application/json'})
# => GET http://sushi.com/nigiri/sake.json?a=1
```

https://lostisland.github.io/faraday/usage/

You can get rid of this `?id=#{tweet_id}` part and use the syntax from that example. Try it ;)